src/core/agent/services/signifyNotificationService.ts

```diff
@@ -218,6 +231,50 @@ class SignifyNotificationService extends AgentService {
     });
     return notificationRecord;
   }
+  async onSignifyOperationStateChanged(
+    callback: ({ record, recordType }: {record: IdentifierShortDetails, recordType: OperationPendingRecordType}) => void
+  ) {
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      const pendingOperations =
+        await this.operationPendingStorage.getAll();
+      if (pendingOperations.length > 0) {
+        for (const pendingOperation of pendingOperations) {
+          const operation = await  this.signifyClient.operations().get(pendingOperation.id);
+          if (operation.done) {
+            switch (pendingOperation.recordType) {
+            case OperationPendingRecordType.IDENTIFIER: {
+              const aid = await this.identifierStorage.getIdentifierMetadata(pendingOperation.recordId);
+              await this.identifierStorage.updateIdentifierMetadata(aid.id, {
+                isPending: false,
+              });
+              callback({
+                recordType: pendingOperation.recordType,
+                record: {
+                  displayName: aid.displayName,
+                  id: aid.id,
+                  signifyName: aid.signifyName,
+                  createdAtUTC: aid.createdAt.toISOString(),
+                  theme: aid.theme,
+                  isPending: false,
+                }
+              });
+              await this.operationPendingStorage.deleteById(pendingOperation.id);
+              break;
+            }
+            default:
+              break;
+            }
+          }
+        }
+      }
+      await new Promise((rs) => {
+        setTimeout(() => {
+          rs(true);
+        }, 20000);
```

Comment:
This is too high. For now lets do 250ms - in the future we can go lower but add backoff